slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/interchange/InterchangeV5Exporter.java

```diff
@@ -102,58 +107,101 @@ private void populateInterchangeData(final JsonGenerator jsonGenerator) {
   }
 
   private void populateValidatorRecord(
-      final Handle h, final Validator validator, final JsonGenerator jsonGenerator)
+      final Handle handle, final Validator validator, final JsonGenerator jsonGenerator)
       throws IOException {
     jsonGenerator.writeStartObject();
     jsonGenerator.writeStringField("pubkey", validator.getPublicKey().toHexString());
-    writeBlocks(h, validator, jsonGenerator);
-    writeAttestations(h, validator, jsonGenerator);
+    final Optional<SigningWatermark> watermark =
+        lowWatermarkDao.findLowWatermarkForValidator(handle, validator.getId());
+    if (watermark.isEmpty()) {
+      LOG.warn("No low watermark available, producing empty export.");
+    } else {
+      writeBlocks(handle, watermark.get(), validator, jsonGenerator);
+      writeAttestations(handle, watermark.get(), validator, jsonGenerator);
+    }
     jsonGenerator.writeEndObject();
   }
 
   private void writeBlocks(
-      final Handle h, final Validator validator, final JsonGenerator jsonGenerator)
+      final Handle handle,
+      final SigningWatermark watermark,
+      final Validator validator,
+      final JsonGenerator jsonGenerator)
       throws IOException {
     jsonGenerator.writeArrayFieldStart("signed_blocks");
 
-    signedBlocksDao
-        .findAllBlockSignedBy(h, validator.getId())
-        .forEach(
-            b -> {
-              final tech.pegasys.web3signer.slashingprotection.interchange.model.SignedBlock
-                  jsonBlock =
-                      new tech.pegasys.web3signer.slashingprotection.interchange.model.SignedBlock(
-                          b.getSlot(), b.getSigningRoot().orElse(null));
-              try {
-                mapper.writeValue(jsonGenerator, jsonBlock);
-              } catch (final IOException e) {
-                throw new UncheckedIOException(
-                    "Failed to construct a signed_blocks entry in json", e);
-              }
-            });
+    if (watermark.getSlot() == null) {
+      LOG.warn(
```

Comment:
Think this is the same as not having a signing record where we don't create a signing record. Perhaps the top level check need to verify that we have all non-null values for the watermark record.